python/ray/serve/api.py

```diff
@@ -1712,47 +1731,79 @@ def from_yaml(cls, str_or_file: Union[str, TextIO]) -> "Application":
                 Serve YAML config files.
 
         Returns:
-            Application: a new application object containing the deployments.
+            Application: a new Application object containing the deployments.
+        """
+        return cls.from_dict(yaml.safe_load(str_or_file))
+
+    def get_ingress(self) -> Optional[Deployment]:
+        """Gets the app's ingress, if one exists.
+
+        The ingress is the single deployment with a non-None route prefix. If more
+        or less than one deployment has a route prefix, no single ingress exists,
+        so returns None.
         """
-        raise NotImplementedError()
+
+        ingress = None
+
+        for deployment in self._deployments.values():
+            if deployment.route_prefix is not None:
+                if ingress is None:
+                    ingress = deployment
+                else:
+                    return None
+
+        return ingress
 
 
 @PublicAPI(stability="alpha")
 def run(
     target: Union[DeploymentNode, Application],
+    blocking: bool = True,
```

Comment:
remove ?

Reply:
`run` now does what `deploy_group` used to, so the flag here allows the REST API to issue a non-blocking deploy request. Alternatively, we could pull out `deploy_group` as a separate helper function in `api.py`. Thoughts?

Reply:
ah, I see. I'd prefer that we don't expose this as a public flag, how about we make it underscore prefixed and don't document it?

Reply:
Sounds good, I prefixed it.